core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableMergeJoin.java

```diff
@@ -101,28 +103,55 @@ public static boolean isMergeJoinSupported(JoinRelType joinType) {
     // Required collation keys can be subset or superset of merge join keys.
     RelCollation collation = required.getTrait(RelCollationTraitDef.INSTANCE);
     List<Integer> reqKeys = RelCollations.ordinals(collation);
-    ImmutableBitSet reqKeySet = ImmutableBitSet.of(reqKeys);
+    // need to copy reqKeys because it is not sortable.
+    reqKeys = new ArrayList<>(reqKeys);
```

Comment:
why do you want to sort it?

Reply:
`isPrefixOrderingNotRequired` sorts its input, and reqKeys is used as this function's parameter below.

But since you have suggested `RelCollations.containsOrderless`, I might not need this anymore.